tests/test_ProgressPrinter.py

```diff
@@ -147,3 +148,47 @@ def test_should_print_false_no_output(ok_small, capsys):
 
     out = capsys.readouterr().out
     assert_equal(out, "")
+
+
+@pytest.mark.parametrize(
+    "routes",
+    [
+        [[1, 2], [3, 4]],  # feasible
+        [[1, 2, 3, 4]],  # infeasible
+    ],
+)
+def test_print_dash_when_subpopulation_is_empty(ok_small, routes, capsys):
+    """
+    Tests that a "-" is printed as cost if one of the subpopulations is empty.
+    """
+    pop = Population(bpd)
+    cost_eval = CostEvaluator(1, 1, 0)
+
+    solution = Solution(ok_small, routes)
+    pop.add(solution, cost_eval)
+
+    stats = Statistics()
+    stats.collect_from(pop, cost_eval)
+    stats.num_iterations = 500  # force printing
+
+    printer = ProgressPrinter(should_print=True)
+    printer.iteration(stats)
+
+    # Population contains either one feasible or one infeasible solution, so
+    # the costs of the empty subpopulation should be printed as "-".
+    out = capsys.readouterr().out
+    assert_("-" in out)
+
+    # Now we add many solutions to make both subpopulations non-empty.
+    rng = RandomNumberGenerator(seed=42)
+    for _ in range(100):
+        pop.add(Solution.make_random(ok_small, rng), cost_eval)
+
+    stats.collect_from(pop, cost_eval)
+    stats.num_iterations = 500  # force printing
+    printer.iteration(stats)
+
+    # The "-" should not be printed anymore, but there should be some output.
+    out = capsys.readouterr().out
+    assert_("-" not in out)
+    assert_(out != "")
```

Comment:
I think this part of the test is somewhat superfluous when we also have the other tests above (which essentially check the same thing). Do you have an example of what the exact output is when one of the subpops is empty? Not necessarily for a test, just for my own curiosity.

Reply:
I think I added it mainly as a sanity check. This is what it looks like when feasible is empty:

```
PyVRP v0.8.0a0

Solving an instance with:
    1 depot
    436 clients
    1 vehicle (1 vehicle type)

                  |       Feasible        |      Infeasible
    Iters    Time |   #      Avg     Best |   #      Avg     Best
      500      6s |   0        -        - |  33 1400514125 348530410
     1000     12s |   0        -        - |  41 17041546427 6233398105
```

Reply:
Yeah that looks nice!